core/templates/dev/head/pages/state_editor/StateHintsEditorDirective.js

```diff
@@ -45,8 +45,8 @@ oppia.directive('stateHintsEditor', [
           $scope.StateHintsService = StateHintsService;
           $scope.activeHintIndex = null;
           $scope.canEdit = GLOBALS.can_edit;
-          if (StateEditorService.getInQuestionMode()) {
-            $scope.canEdit = true;
+          if (StateEditorService.isInQuestionMode()) {
+            $scope.canEdit = EditabilityService.isEditable();
```

Comment:
Any chance we can just use isEditable() everywhere (even for the non-question case) and avoid the branching?

Reply:
Yeah, there is a check in ExplorationEditor.js that sets EditabilityService.editable only when this flag is true, will modify this.

Reply:
Done.